dbms/src/Storages/DeltaMerge/ReadThread/CPU.cpp

```diff
@@ -0,0 +1,112 @@
+#include <Common/Exception.h>
+#include <Poco/DirectoryIterator.h>
+#include <Poco/File.h>
+#include <Poco/Logger.h>
+#include <Storages/DeltaMerge/ReadThread/CPU.h>
+#include <common/logger_useful.h>
+
+#include <exception>
+#include <string>
+
+namespace DB::DM
+{
+// In Linux a numa node is represented by a device directory, such as '/sys/devices/system/node/node0', '/sys/devices/system/node/node01'.
+static inline bool isNodeDir(const std::string & name)
+{
+    return name.size() > 4 && name.substr(0, 4) == "node" && std::all_of(name.begin() + 4, name.end(), [](unsigned char c) { return std::isdigit(c); });
```

Comment:
I think using a regex is better. And we definitely need to do an exception protection caused by illegal `name`.

Reply:
These functions restrict the name to `node[digits]` and `cpu[digits]`. I think this is more clarify and strict. 
No illegal string can cause these functions throw a exception after we check the size of the string.

The advantage of regex is flexible but is not required here.